symDiskCache.py

```diff
@@ -0,0 +1,97 @@
+from symUtil import mkdir_p
+import cPickle as pickle
+import os
+from symLogging import LogDebug
+from collections import deque
+
+# Manage in disk symbol cache
+class SymDiskCache(object):
+  def __init__(self, cachePath, maxNumberOfFiles):
+    mkdir_p(cachePath)
+    self.cachePath = cachePath
+    self.maxNumberOfFiles = maxNumberOfFiles
+
+    self.entryCount, fileList = self._Walk(cachePath)
+
+    # sort the symbol relative to the last access time
+    fileList.sort(cmp=lambda a,b: cmp(a[1], b[1]))
+
+    # filter out access times
+    self.lru = deque(map(lambda x: x[0], fileList))
+
+    self.CheckCacheSize()
+
+  def Fetch(self, filename):
+    r"fetch a symbol file from the cache"
+    path = self.absPath(filename)
+    symbolInfo = None
+    if os.path.exists(path):
+      with open(path, 'r') as f:
+        symbolInfo = pickle.load(f)
+        self.Use(filename)
+    return symbolInfo
+
+  def Store(self, symbolInfo, filename):
+    r"""Store a symbol file in the cache.
+
+    The caller must call use(filename) before this
+    method.
+    """
+    path = self.absPath(filename)
+    dirname = os.path.dirname(path)
+    mkdir_p(dirname)
+    self.Add(filename)
+
+    with open(path, 'w') as f:
+      pickle.dump(symbolInfo, f, pickle.HIGHEST_PROTOCOL)
+
+    self.entryCount += 1
+    self.CheckCacheSize()
+
+  def Use(self, filename):
+    r"Mark the cached sym file as recently used."
+    normPath = os.path.normpath(filename)
+    path = self.absPath(normPath)
+    try:
+      self.lru.remove(normPath)
+    except ValueError:
+      pass
+    else:
+      self.lru.append(normPath)
+
+  def Add(self, filename):
```

Comment:
They are two operations because I want to keep the LRU consistent with memory cache, so when a memory cache entryis hit, I call `Use` to update the list, but it makes sense only if the element is there. `Add` is the opposite, add a element which is not there.